clib/cThread.ml

```diff
@@ -107,3 +107,18 @@ let mask_sigalrm f x =
 
 let create f x =
   Thread.create (mask_sigalrm f) x
+
+(*
+  Atomic mutex lock taken from https://gitlab.com/gadmm/memprof-limits/-/blob/master/src/thread_map.ml#L23-34
+  Critical sections :
+   - Mutex.lock does not poll on leaving the blocking section
+     since 4.12.
+   - Never inline, to avoid theoretically-possible reorderings with
+     flambda.
+     (workaround to the lack of masking)
+*)
+let[@inline never] with_lock m ~scope =
+  let () = Mutex.lock m (* BEGIN ATOMIC *) in
+  match (* END ATOMIC *) scope () with
```

Comment:
Given you want to be that precise, can you put this in administrative normal form (hoist the `scope ()` value)?
I don't know what your comment means otherwise (without already knowing it).

Reply:
I mean
```
  match
    (*here*)
    let res = scope () in
   (* or here *)
    res
  with
```

Reply:
(I admit I'm not a super expert, but if the comment does not help me, I'm unsure it will help many readers out there)

Reply:
Yeah, I´m also not exactly sure what these comments mean (@gadmm  wrote them). Maybe he can comment on what they indicate exactly?

Reply:
It is true that it is not entirely precise; the beginning of the atomic scope is inside the C function that defines Mutex.lock. Would it be fine to simply remove the comments?

Reply:
yes

Reply:
I like the comments tho, I got their purpose and they are useful.